src/node/interfaces.cpp

```diff
@@ -395,9 +396,42 @@ class NodeImpl : public Node
     {
         m_context = context;
     }
+    double getSnapshotProgress() override { return m_snapshot_progress.load(); }
+    bool snapshotLoad(const std::string& path_string) override
+    {
+        const fs::path path = fs::u8path(path_string);
+        if (!fs::exists(path)) { return false; }
+
+        AutoFile afile{fsbridge::fopen(path, "rb")};
+        if (afile.IsNull()) { return false; }
+
+        SnapshotMetadata metadata;
+        try {
+            afile >> metadata;
+        } catch (const std::exception& e) { return false; }
+
+        const uint256& base_blockhash = metadata.m_base_blockhash;
+
+        if (!m_context->chainman) { return false; }
+
+        ChainstateManager& chainman = *m_context->chainman;
+        CBlockIndex* snapshot_start_block = nullptr;
+
+        // Wait for the block to appear in the block index
+        //TODO: remove this once another method is implemented
+        constexpr int max_wait_seconds = 600; // 10 minutes
+        for (int i = 0; i < max_wait_seconds; ++i) {
```

Comment:
We shouldn't have a blocking component here. If there are any issues we should just fail.